packages/jest-runtime/src/index.ts

```diff
@@ -1506,10 +1506,8 @@ export default class Runtime {
     filename: string,
     options?: InternalModuleOptions,
   ): string {
-    const source = this.readFile(filename);
```

Comment:
hmm, `readFile` looks in `cacheFS`, same as the transformer. does this actually change anything? it's the same read operation cached at slightly different times?

https://github.com/facebook/jest/blob/c791d97114024391b1894d9923688c0a0216cc4f/packages/jest-runtime/src/index.ts#L2308-L2318

Reply:
If I'm wrong and it's _not_ the same cache, we should fix that

Reply:
Yep, [it's the same backing `cacheFS` instance](https://github.com/facebook/jest/blob/main/packages/jest-runner/src/runTest.ts#L106-L107) :+1: 

> does this actually change anything

Err, you're absolutely correct. This needs to go another layer deeper:
* `_transformAndBuildScript()` [still wants to read the source file](https://github.com/facebook/jest/blob/main/packages/jest-runner/src/runTest.ts#L106-L107tSc)
* [The only usage of the "original code" is here, in the return value](https://github.com/facebook/jest/blob/main/packages/jest-runner/src/runTest.ts#L106-L107tSc)
* The only usage of the "original code" is [here, in the coverage reporter](https://github.com/facebook/jest/blob/50bab21a8e88215061228e4197bfe632a529692c/packages/jest-reporters/src/CoverageReporter.ts#L464)

TL;DR: This PR needs more work to avoid the unnecessary `readFile`, as it depends on whether or not coverage reporting is going on.
Sorry about that!

Reply:
Thanks for checking! I'm super happy somebody is looking into these performance improvements 🙂 death by a thousand cuts and all that